tests/unit/test_snap.py

```diff
@@ -697,6 +700,141 @@ def test_request_raw_bad_response_raises_snapapierror(self):
+                                "status": "Doing",
+                                "tasks": [
+                                    {
+                                        "id": "1029",
+                                        "kind": "run-hook",
+                                        "summary": 'Run configure hook of "test" snap',
+                                        "status": "Doing",
+                                        "progress": {"label": "", "done": 1, "total": 1},
+                                        "spawn-time": "2024-11-28T20:02:47.498399651+00:00",
+                                        "data": {"affected-snaps": ["test"]},
+                                    }
+                                ],
+                                "ready": False,
+                                "spawn-time": "2024-11-28T20:02:47.49842583+00:00",
+                            },
+                        }
+                    ).encode("utf-8")
+                )
+            if method == "GET" and path == "changes/97" and change_finished:
+                return io.BytesIO(
+                    json.dumps(
+                        {
+                            "type": "sync",
+                            "status-code": 200,
+                            "status": "OK",
+                            "result": {
+                                "id": "98",
+                                "kind": "configure-snap",
+                                "summary": 'Change configuration of "test" snap',
+                                "status": "Done",
+                                "tasks": [
+                                    {
+                                        "id": "1030",
+                                        "kind": "run-hook",
+                                        "summary": 'Run configure hook of "test" snap',
+                                        "status": "Done",
+                                        "progress": {"label": "", "done": 1, "total": 1},
+                                        "spawn-time": "2024-11-28T20:06:41.415929854+00:00",
+                                        "ready-time": "2024-11-28T20:06:41.797437537+00:00",
+                                        "data": {"affected-snaps": ["test"]},
+                                    }
+                                ],
+                                "ready": True,
+                                "spawn-time": "2024-11-28T20:06:41.415955681+00:00",
+                                "ready-time": "2024-11-28T20:06:41.797440022+00:00",
+                            },
+                        }
+                    ).encode("utf-8")
+                )
+            raise RuntimeError("unknown request")
+
+        client = snap.SnapClient()
+        with patch.object(client, "_request_raw", _request_raw), patch.object(time, "sleep"):
+            client._put_snap_conf("test", {"foo": "bar"})
```

Comment:
I guess ideally we'd call the public `Snap.set` instead of the private `SnapClient._put_snap_client` method, but that's not a blocker